pandas/tslib.pyx

```diff
@@ -904,10 +909,9 @@ cpdef object get_value_box(ndarray arr, object loc):
 
 
 # Add the min and max fields at the class level
-# These are defined as magic numbers due to strange
-# wraparound behavior when using the true int64 lower boundary
-cdef int64_t _NS_LOWER_BOUND = -9223285636854775000LL
-cdef int64_t _NS_UPPER_BOUND = 9223372036854775807LL
+# INT64_MIN is reserved for NaT
+cdef int64_t _NS_LOWER_BOUND = INT64_MIN + 1
```

Comment:
On further look, I'll change the lower bound back - in our compat with python datetime we assume in several places that the min value has a 0 nanosecond unit - I think it _could_ be worked around, but fairly invasive for another 800 ns of span.